packages/server/src/modules/documents/helpers/match-transaction.helper.ts

```diff
@@ -0,0 +1,36 @@
+import type { Injector } from 'graphql-modules';
+import { LLMProvider, type DocumentData } from '@modules/app-providers/llm.js';
+import { TransactionsProvider } from '@modules/transactions/providers/transactions.provider.js';
+
+/**
+ * Attempts to match a document to unmatched transactions in the system
+ * @param injector The GraphQL modules injector
+ * @param document The document data to match
+ * @returns The ID of the matched transaction, or null if no match found
+ */
+export async function matchTransactionToDocument(
+  injector: Injector,
+  document: DocumentData,
+  context: GraphQLModules.ModuleContext,
+): Promise<string | null> {
+  try {
+    // Get unmatched transactions (those with a charge_id but no matching document)
+    const unmatchedTransactions = await injector
+      .get(TransactionsProvider)
+      .getTransactionsWithChargeButNoDocument(context.adminContext.defaultAdminBusinessId);
+
+    if (!unmatchedTransactions || unmatchedTransactions.length === 0) {
+      return null;
+    }
```

Comment:
Thoughts:
Currently the matching options are based solely on doc-less transactions.
Maybe we should limit the options by some date / amount+currency proximity
If document counterparty exists, that as well.
Maybe not limit the options, but enrich them with transactions that DO have docs but info matches very well

I mean, it ends up with converting parts the AI job to our code, but given the poor current results maybe AI could do better after some digestion + focus on sorting the options by "match potential"